src/crypto/aead.rs

```diff
@@ -266,23 +255,19 @@ impl AeadSchemeInterface for Aes128Gcm {
         let key = enum_variant!(key, AeadKey::Aes128GcmKey);
         let nonce = enum_variant!(nonce, AeadNonce::Aes128GcmNonce);
 
+        let (data, tag_out) = plaintext.split_at_mut(plaintext.len() - AES_128_GCM_TAG_SIZE);;
```

Comment:
Note: In real code, you'd need to change this to prevent problems if `plaintext.len()` is shorter than `AES_128_GCM_TAG_SIZE`.

However, I instead encourage you to try to replace your abstraction over the *ring* AEAD API and instead just use the *ring* AEAD API directly in the higher-level code. In *ring* 0.16 we added [`seal_in_place_append_tag`](https://briansmith.org/rustdoc/ring/aead/struct.LessSafeKey.html#method.seal_in_place_append_tag) which is a safer and easier to understand API than your current `seal()` API and the old *ring* API (based on feedback from *ring* users).